api/handlers/dataexplorerhandler.py

```diff
@@ -305,7 +305,39 @@ def _parse_request(self, request_type='search'):
         for f in filters:
             if f.get('terms'):
                 for k,v in f['terms'].iteritems():
-                    modified_filters.append({'terms': {k+'.raw': v}})
+                    if "null" in v:
+                        try:
+                            v.remove("null")
+                        except AttributeError:
+                            v = None
+                        null_filter = {
+                            'bool': {
+                                'should': [
```

Comment:
Reading the docs again it looks like since this is in a `filter` context and not a `query` context, `should` means "must match at least one". 

>If the bool query is a filter context or has neither must or filter then at least one of the should queries must match a document for it to match the bool query.

https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-bool-query.html#query-dsl-bool-query

Which doesn't actually make a difference in this context but ...
![image](https://user-images.githubusercontent.com/5702596/28734217-772778dc-73a5-11e7-911e-6f707aefd796.png)